### PR TITLE
javascript editor

### PR DESCRIPTION
**Pull request by breakfast-for-dinner**
there is still a lot to be done but other ppl can help out with that stuff !!!
<hr>
the advanced editor is visible through the editor with the "advanced" button at the top.

for previewing: you can create a new sprite using this syntax:

```js
let name = new sprite(type, x, y, size)
```

the only available options for `sprite.type` are `jonathan` and `dog` at this point in time

you can draw sprites using the `draw` command on the sprite.

### CODE REVIEW

File: editor/cool/canvas.html [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            width: 670px;
+            height: 160px;
+        }
+    </style>
+</head>
+<body>
+    <canvas width="700" height="400"></canvas>
+    <textarea id="projecttext" spellcheck="false"></textarea>
+    <script>
+        var canvas = document.getElementsByTagName("canvas")[0].getContext("2d");
```
Comment (dumorando) [inline editor/cool/canvas.html:36]: change var to let/const

**Comment by dumorando** [comment]
other then that lgtm

**Review by dumorando — APPROVED** [review]

**Comment by mariocraft987** [comment]
I aprove this

**Comment by mariocraft987** [comment]
like the technic version of lego

**Comment by Bashamega** [comment]
@mariocraft987 Please let me have access to see the preview.

**Comment by mariocraft987** [comment]
Done!

**Comment by breakfast-for-dinner** [comment]
@mariocraft987 could you give me permissions as well? (for the future) username is `qwertyy-dev`

**Comment by mariocraft987** [comment]
umm...

**Comment by Bashamega** [comment]
> Done!

thanks

**Comment by Bashamega** [comment]
It is not working @mariocraft987 
image

**Comment by breakfast-for-dinner** [comment]
> Screenshot 2024-06-26 201901

oh. that's alright since it's my deployment anyway, give basha the controls

**Comment by mariocraft987** [comment]
done

**Comment by Bashamega** [comment]
I think it is good but you should switch from the lame editor to [Manaco Editor](https://microsoft.github.io/monaco-editor/). It will give you the feel of vscode editor.

**Comment by dumorando** [comment]
lame?? lol

monaco is quite hard to embed in a vanilla JS site so i'd personally recommend [ace](https://ace.c9.io)

**Comment by breakfast-for-dinner** [comment]
> lame editor

by that do you mean my editor or the block editor?

**Comment by dumorando** [comment]
i assume he means the text box code editor

**Comment by Bashamega** [comment]
> lame?? lol
> 
> monaco is quite hard to embed in a vanilla JS site so i'd personally recommend [ace](https://ace.c9.io)

yes editor doesn't have like code highlighting like that

**Comment by mariocraft987** [comment]
no predicting text☠️

**Comment by dumorando** [comment]
i'm just gonna merge it this feedback is kinda going nowhere

**Comment by Bashamega** [comment]
> no predicting text☠️

Sorry, i meant it could have a better ui

**Comment by breakfast-for-dinner** [comment]
> no predicting text☠️



> > yes editor doesn't have like code highlighting like that

i want to reiterate that there is still a lot to be done

**Comment by dumorando** [comment]
either way we can add nicer code editor in the future it's not like we need to add it right now

**Pull request merged**